packages/core/lib/debug/printer.js

```diff
@@ -171,13 +171,31 @@ class DebugPrinter {
     const colorizedLines = splitLines(colorizedSource);
 
     this.config.logger.log("");
+    // We clone range though only "lines" is used here.
+    let rangePrintout = JSON.parse(JSON.stringify(range));
```

Comment:
Please don't perform deep clones this way.  This is a hack and, if there were truly a reason you needed to resort to it, it ought to be labeled as such.  You generally shouldn't need to perform deep cloning at all though, and you certainly don't need to here.  If you did need to, I would suggest importing `lodash.clonedeep` to do it -- but, again, there's absolutely no reason to perform a clone here, whether deep or shallow.

As best I can tell, the reason for this clone is... to preserve the existing structure of variables that's used?  That's not something you need to preserve.  Rather than performing a deep clone to preserve the existing structure of variables used, you should change the variables used to something more appropriate for the new situation.  You should just be extracting and using the relevant information, not cloning irrelevant information.

(Btw: You can even do this without using any mutation.  Avoiding mutation is nice when possible as it makes the meaning of variables clearer.  I mean you don't want to go too hard on that, don't avoid mutation if that will make things difficult, but...)

In a more minor issue that occurs on this line, I must also point out that this variable's name is confusing.  You named it `rangePrintout`; that sounds like a printout for a range, when you meant a range for a printout.  If you keep this variable (although I imagine there's a good chance you'll remove it), you'd probably want to rename it to `printoutRange` or `rangeForPrintout`.  Remember, English is a head-second language; if you write a noun-noun compound, the first noun will be read as modifying the second, not vice-versa.  Now you will often see variables named with modifiers coming after the noun, but importantly, in those cases, the modifiers are *adjectives*, not other nouns, which makes the result unambiguous regardless of order.  With two nouns, though, you can't use part of speech to disambiguate, so you have to use order; thus the first noun will be read as modfiying the second.  You'll want to keep that in mind when naming variables.

Reply:
I consider to keep the variable `rangePrintout` and rename it to `printoutRange`.  This variable will contain `sourceRange.lines` only.  The initial value of this variable is `range.lines` from input and is set to the end of source code if the invalid byte-offset is detected.

Reply:
That's really needlessly indirect, don't you think?  If it's only going to contain `lines`, shouldn't you just use `lines`?  Why would you have a variable just to contain it?

Reply:
Oops, it seems I misunderstood @dongmingh's comment here -- he meant contain *as the value*, not contain *as a subfield*.  Sorry about that!  Yes, this approach makes sense given what was actually meant.